src/test/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/servicebus/ProcessedEnvelopeNotifierTest.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.microsoft.azure.servicebus.Message;
+import com.microsoft.azure.servicebus.QueueClient;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProcessedEnvelopeNotifierTest {
+
+    private ProcessedEnvelopeNotifier notifier;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Mock
+    private QueueClient queueClient;
+
+    @Before
+    public void setUp() {
+        notifier = new ProcessedEnvelopeNotifier(queueClient, objectMapper);
+    }
+
+    @Test
+    public void notify_should_send_message_with_right_content() throws Exception {
+        // given
+        String envelopeId = UUID.randomUUID().toString();
+
+        // when
+        notifier.notify(envelopeId);
+
+        // then
+        ArgumentCaptor<Message> messageCaptor = ArgumentCaptor.forClass(Message.class);
+        verify(queueClient).send(messageCaptor.capture());
+        Message message = messageCaptor.getValue();
+
+        assertThat(message.getMessageId()).isEqualTo(envelopeId);
+        assertThat(message.getContentType()).isEqualTo("application/json");
+
+        String messageBodyJson = new String(message.getBody());
+        String expectedMessageBodyJson = String.format("{\"id\":\"%s\"}", envelopeId);
+        JSONAssert.assertEquals(expectedMessageBodyJson, messageBodyJson, JSONCompareMode.LENIENT);
+    }
+
+    @Test
+    public void notify_should_throw_exception_when_queue_client_fails() throws Exception {
+        ServiceBusException exceptionToThrow = new ServiceBusException(true, "test exception");
+        willThrow(exceptionToThrow).given(queueClient).send(any());
+
+        assertThatThrownBy(() -> notifier.notify("envelopeId123"))
+            .isInstanceOf(MessageSendingException.class)
+            .hasMessage("An error occurred when trying to send a message to processed envelopes queue")
```

Comment:
it may be worth including envelope id in the exception message imho

Reply:
The Id will is included in the error log.

Reply:
Intereseting, I assume it's already existing code, as it's not here.
This leads me to questions:
How is this exception handled handle?
And, perhaps more importantly, what happens to the envelope queue message?

Reply:
Well, when we log errors, we add some context to the exceptions. And that's what happens here.

If we fail to send the message to `processed-envelopes` queue, we assume the processing of the incoming message has failed. This case is not much different than other kinds of message processing failures, such as when CCD has done it's job but failed to return response.

Reply:
I don't think it will fail on the next attempt. We're not free from unwanted reprocessing. Here are some examples of what can happen:
- CCD completes the operation, but the orchestrator is not informed about the success (timeout, connection issue, etc.)
- CCD completes the operation, but the orchestrator fails to complete the incoming message

Having thought about it, our case is the worst, because if, for example, the output queue is down, we'll keep creating stuff in CCD. I'll dead-letter it, but we still have a bigger problem to solve here.